alert-common/src/main/java/com/synopsys/integration/alert/common/persistence/util/FilePersistenceUtil.java

```diff
@@ -49,17 +54,35 @@ public FilePersistenceUtil(final AlertProperties alertProperties, final Gson gso
         }
         this.parentDataDirectory = new File(dataDirectory);
         this.secretsDirectory = new File(alertProperties.getAlertSecretsDir());
+        this.uploadsDirectory = new File(parentDataDirectory, "uploads");
     }
 
     public void writeToFile(final String fileName, final String content) throws IOException {
-        FileUtils.write(createFile(fileName), content, ENCODING);
+        writeToFile(createFile(fileName), content.getBytes(StandardCharsets.UTF_8));
+    }
+
+    public void writeToFile(File destination, byte[] data) throws IOException {
+        // destination is a path to a file.  Make sure all parent directories exist before writing.
+        destination.getParentFile().mkdirs();
+        Files.write(destination.toPath(), data, StandardOpenOption.CREATE, StandardOpenOption.WRITE, StandardOpenOption.TRUNCATE_EXISTING);
+    }
+
+    public void writeToFile(File destination, InputStream inputStream) throws IOException {
```

Comment:
Should this method technically be called copy as we're copying a file?

Reply:
The inputStream that we are "copying" into the destination file does not have to be a file that exists. It could be a stream in memory. We are just putting the data from the stream into the destination file. I think the method is named appropriately.

Reply:
James is correct here.  Files calls the method copy because you are copying the contents of the input stream, whether it exists or not.